### PR TITLE
feat: Add substrate executor module

### PR DESCRIPTION
**Pull request by tcar121293**
Add substrate executor module
## Description
Added substrate executor module
Added substrate proposal struct

## Related Issue Or Context

Closes: #25 

## How Has This Been Tested? Testing details.
Will be tested with e2e tests

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation

## Checklist:

- [ ] I have commented my code, particularly in hard-to-understand areas.
- [x] I have ensured that all acceptance criteria (or expected behavior) from issue are met
- [ ] I have updated the documentation locally and in docs.
- [ ] I have added tests to cover my changes.
- [x] I have ensured that all the checks are passing and green, I've signed the CLA bot

### CODE REVIEW

File: chains/substrate/executor/executor.go [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+func (e *Executor) Execute(msgs []*message.Message) error {
+	proposals := make([]*proposal.Proposal, 0)
+	for _, m := range msgs {
+		prop, err := e.mh.HandleMessage(m)
+		if err != nil {
+			return err
+		}
+
+		isExecuted := e.bridge.IsProposalExecuted(prop)
+
```
Comment (mpetrun5) [inline chains/substrate/executor/executor.go:80]:
```suggestion
```

File: chains/substrate/executor/proposal/proposal.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
+package proposal
+
+import (
+	"github.com/ChainSafe/chainbridge-core/relayer/message"
+	"github.com/ChainSafe/chainbridge-core/types"
+)
+
+func NewProposal(source, destination uint8, depositNonce uint64, resourceId types.ResourceID, data []byte) *Proposal {
+
```
Comment (mpetrun5) [inline chains/substrate/executor/proposal/proposal.go:9]:
```suggestion
```

**Review by mpetrun5 — COMMENTED** [review]
It does seem okay, but without e2e tests it is hard to say anything

**Comment by tcar121293** [comment]
> It does seem okay, but without e2e tests it is hard to say anything

I Agree, will change the PR into draft PR until we'll have e2e tests ready

**Comment by mpetrun5** [comment]
Merging this and the fixes will be done in e2e branch.

**Pull request merged**